src/librustc/mir/repr.rs

```diff
@@ -890,6 +890,13 @@ pub struct Constant<'tcx> {
     pub literal: Literal<'tcx>,
 }
 
+#[derive(Clone, Debug, RustcEncodable, RustcDecodable)]
```

Comment:
I think an explicit `Debug` implementation based on `fmt_const_val` below is necessary (mir pretty and graphviz outputs depend on that).